### PR TITLE
CSRS 292:

### PR DESCRIPTION
**Pull request by ryder-cayden**
Tweak the file manager service as per CSRS specs
Update Title from Content Type parameter

# Description

This PR includes the following proposed change(s):

- {List all the changes, if possible add the jira ticket #}

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [ ] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/Csrs.Interfaces.SharePoint/SharePointFileManager.cs [hunk 1]
```diff
@@ -872,10 +928,77 @@ public async Task<string> UploadFile(string fileName, string listTitle, string f
...
+                endpointRequest.Headers.Add("odata-version", "3.0");
+                endpointRequest.Content = strContent;
+
+                //ByteArrayContent byteArrayContent = new ByteArrayContent(data);
+                //byteArrayContent.Headers.Add(@"content-length", data.Length.ToString());
+                //endpointRequest.Content = byteArrayContent;
+
+                // make the request.
+                var response = await _Client.SendAsync(endpointRequest);
+                var streamData = response.Content.ReadAsStringAsync().Result;
```
Comment (pbolduc) [inline src/backend/Csrs.Interfaces.SharePoint/SharePointFileManager.cs:997]: Use `await`

File: src/backend/Csrs.Interfaces.SharePoint/SharePointFileManager.cs [hunk 2]
```diff
@@ -872,10 +928,77 @@ public async Task<string> UploadFile(string fileName, string listTitle, string f
...
                 if (response.StatusCode == HttpStatusCode.OK)
                 {
                     result = fileName;
+
+                    return await UpdateListItemFields(listItemData, listTitle, contentType, fileName); ;
+                }
+                else
+                {
+                    string _responseContent = null;
+                    var ex = new SharePointRestException(string.Format("Operation returned an invalid status code '{0}'", response.StatusCode));
```
Comment (pbolduc) [inline src/backend/Csrs.Interfaces.SharePoint/SharePointFileManager.cs:944]: Use string interpolation

**Review by pbolduc — CHANGES_REQUESTED** [review]

File: src/backend/Csrs.Interfaces.SharePoint/SharePointFileManager.cs (hunk 2 above)
Comment (ryder-cayden) [reply src/backend/Csrs.Interfaces.SharePoint/SharePointFileManager.cs]: changed all the SharePointRestException to use string interpolation

File: src/backend/Csrs.Interfaces.SharePoint/SharePointFileManager.cs (hunk 1 above)
Comment (ryder-cayden) [reply src/backend/Csrs.Interfaces.SharePoint/SharePointFileManager.cs]: updated to await

**Comment by ryder-cayden** [comment]
updated code based on review by Phil

**Pull request merged**